lib/postprocessing/image.js

```diff
@@ -87,23 +85,49 @@ function hasExtraInstructions(instructions) {
 }
 
 async function getImageMetadata(file) {
-    const start = new Date();
     try {
-        // [1x1+0+0] is a trick to avoid reading all pixels into memory, improving time for large images
-        const result = await gm(`${file}[1x1+0+0]`).write("json:-");
-        const json = JSON.parse(result.toString());
+        // suffix attributes with `#` to disable human-readable parsing if exiftool not used with `-n` flag
+        const attributes = [
```

Comment:
I only extract the metadata we need for post-processing handling.

Reply:
I would just extract all metadata. As the file is locally available here, there is not much to win by limiting it, and it probably reads everything anyway.

Having all metadata makes future changes that need more metadata fields easier.

Reply:
We don't want that. To get the quality estimate, I would have to force the tool to request all metadata - those stored but also those which must be calculated, like the quality. Also, this forces the tool to generate fields that would not normally be generated (calculated fields, nested fields, parsing of fields stored in hexa). This incurs a performance hit. Also, the metadata object ends up being way larger than what we had before. 

So here I limited it to fields we use already on purpose, to mitigate this. Having `JPEGQualityEstimate` explicitly listed as field here will force its parsing, and only its parsing - so I don't need to force exiftool to parse all metadata and the object we carry around stays smaller in size.

Reply:
`JPEGQualityEstimate ` is an extra tag, see https://exiftool.org/TagNames/Extra.html.

To get all metadata, we would need the `RequestAll` flag set to 3. Again, this value must set to 3 in order to grab the quality estimate. From [documentation](https://exiftool.org/ExifTool.html): Flag to request all tags to be extracted. This causes some tags to be generated which normally would not be unless specifically requested (by passing the tag name to ImageInfo or ExtractInfo). May be set to 2 or 3 to enable generation of some additional tags as mentioned in the tag name documentation.

Reply:
1. you already ask for the `JPEGQualityEstimate` flag - how would asking for all metadata change that part?
2. IIUC with imagemagick this quality was read from a metadata that is usually added to the file. We should definitely avoid calculating it, that doesn't really bring any benefit. Maybe there is an option to tell exiftool to not do that? That would definitely be preferred to keep overall behavior as it is today.
3. As mentioned in the Slack discussion, maybe we could drop [this check](https://github.com/adobe/asset-compute-sdk/blob/65541a51b2df1fd63740c71b877ae2d477ad8f44/lib/postprocessing/image.js#L175) altogether if we think quality is not a reliable metadata. And just assume if it's jpeg, then whoever created it did respect the `instructions.quality`. Just need to make sure this is true if the source file is jpeg already. (To be clear: I forgot that we check for the quality metadata here.).

Reply:
1. Because `JPEGQualityEstimate` is named explicitly, only this flag is calculated (again, it is *calculated*, not read). Asking for all would trigger calculation for all calculated flags. 
2. No, the hypothesis is wrong. Imagemagick computed it. This depends on the tool that created the file. The metadata will be written while compressing, but removed afterwards usually. This is part of to the [JFIF (JPEG) spec](https://www.w3.org/Graphics/JPEG/jfif3.pdf): **the quality number (0-100) is not stored in the image header**. For other formats see [imagemagick's quality param](https://imagemagick.org/script/command-line-options.php#quality). It always depends on compression, and is never written as a metadata. Photoshop writes also a quality value, which has nothing to do with that one (it's another scale, max value is 12).
3. Who should make the call to drop it?

Reply:
> Who should make the call to drop it?

Analyze the situation. Do we have any cases where this could be an issue if we drop?

Reply:
The problem I have with the jpeg quality value is that it's useless at best, and misguiding at worst. The JPEG standard does not actually define the "quality" factor, only how the decompression works. The Discrete Cosine Transformation is usually what's used here, but either through implementation details, or through actual algorithmic differences (see https://en.wikipedia.org/wiki/JPEG#JPEG_compression).

In other words, the quality factor is going to have different meanings for different encoders. Different encoders will have different scales and different meanings for these scales, which causes confusion even for the end-users (see https://photo.stackexchange.com/questions/11544/why-does-jpeg-quality-go-up-to-twelve - Photoshop uses a scale from 0 to 12, whereas other encoders have a scale from 0 to 100).

So if a developer wants to make any actual use of these values, they'll need to use the same exact encoder that was used originally with this quality value. If they use a jpeg that was encoded on a scale from 0 to 12, and pass whichever value from it into an encoder that's working on a 0-100 scale, the results won't be what they expect at all. And of course, the encoder originally used isn't part of the metadata.

Long story short, I strongly believe that having the quality metadata exposed can cause more harm than good.